x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_lifecycle/general_data/modal/modal.tsx

```diff
@@ -116,6 +120,23 @@ export function EditLifecycleModal({
     (definition.effective_lifecycle as IngestStreamLifecycleDSL).dsl?.data_retention ??
     `${DEFAULT_RETENTION_VALUE}${DEFAULT_RETENTION_UNIT.value}`;
 
+  const copyCodeContent = React.useMemo(() => {
+    if (isDisabledLifecycle(lifecycle) || isErrorLifecycle(lifecycle)) {
+      return '';
+    }
+
+    const body = buildLifecycleSaveRequestPayload(
+      definition,
+      isInheritToggleOn ? { inherit: {} } : lifecycle
+    );
```

Comment:
nit: I think this logic which `"streamsAppModalFooterButton"` also uses, could be extracted into a reusable function in the same component. Then we may not need a separate function for `buildLifecycleSaveRequestPayload` as it's not doing much.